src/cli/commands/install.js

```diff
@@ -640,6 +687,71 @@ export class Install {
 
     return request;
   }
+
+  /**
+   * Check for updates every day and output a nag message if there's a newer version.
+   */
+
+  checkUpdate() {
+    if (!process.stdout.isTTY || isCI) {
+      // don't show upgrade dialog on CI or non-TTY terminals
+      return;
+    }
+
+    // only check for updates once a day
+    const lastUpdateCheck = Number(this.config.getOption('lastUpdateCheck')) || 0;
+    if (lastUpdateCheck && Date.now() - lastUpdateCheck < ONE_DAY) {
+      return;
+    }
+
+    // don't bug for updates on tagged releases
+    if (YARN_VERSION.indexOf('-') >= 0) {
+      return;
+    }
+
+    this._checkUpdate().catch(() => {
+      // swallow errors
+    });
+  }
+
+  async _checkUpdate(): Promise<void> {
+    let latestVersion = await this.config.requestManager.request({
+      url: 'https://yarnpkg.com/latest-version',
+    });
+    invariant(typeof latestVersion === 'string', 'expected string');
+    latestVersion = latestVersion.trim();
+    if (!semver.valid(latestVersion)) {
+      return;
+    }
+
+    // ensure we only check for updates periodically
+    this.config.registries.yarn.saveHomeConfig({
+      lastUpdateCheck: Date.now(),
+    });
+
+    if (semver.gt(latestVersion, YARN_VERSION)) {
+      this.maybeOutputUpdate = () => {
+        this.reporter.warn(this.reporter.lang('yarnOutdated', latestVersion, YARN_VERSION));
+
+        const command = getUpdateCommand();
+        if (command) {
+          this.reporter.info(this.reporter.lang('yarnOutdatedCommand', command));
+        } else {
+          const installer = getUpdateInstaller();
+          if (installer) {
+            this.reporter.info(this.reporter.lang('yarnOutdatedInstaller', installer));
```

Comment:
Optional, but can we use something like [boxen](https://github.com/sindresorhus/boxen) (or just render the ANSI codes ourself) to render a fancy box around it, like `update-notifier` does?